File: DynamicKey/AgoraDynamicKey/lua-resty/lib/resty/agoradynamickey/signaling_token.lua
local ngx = ngx
local tostring = tostring
local concat = table.concat
local md5 = ngx.md5

local _M = {}


local function generate_signaling_token(account, app_id, app_certificate, expired_ts_in_seconds)
    local version = "1"
    local expired = tostring(expired_ts_in_seconds)
    local content = concat({account, app_id, app_certificate, expired})
    local md5sum = md5(content)
    local result = concat({version, app_id, expired, md5sum}, ":")
    return result
end


_M.generate_signaling_token = generate_signaling_token

return _M

File: DynamicKey/AgoraDynamicKey/go/src/SignalingToken/SignalingToken.go
package SignalingToken

import (
	"fmt"
	"crypto/md5"
	"encoding/hex"
)

func GenerateSignalingToken(account, appID, appCertificate string, expiredTsInSeconds uint32) (string) {

	version := "1"
	expired := fmt.Sprint(expiredTsInSeconds)
	content := account + appID + appCertificate + expired

	hasher := md5.New()
    hasher.Write([]byte(content))
	md5sum := hex.EncodeToString(hasher.Sum(nil))

	result := fmt.Sprintf("%s:%s:%s:%s", version, appID, expired, md5sum)
	return result
}
